source/docs/software/networktables/nt4-migration-guide.rst

```diff
@@ -0,0 +1,165 @@
+Migrating from NetworkTables 3.0 to NetworkTables 4.0
+=====================================================
+
+NetworkTables 4.0 (new for 2023) has a number of significant API breaking changes from NetworkTables 3.0, the version of NetworkTables used from 2016-2022.
+
+NetworkTableEntry
+-----------------
+
+While ``NetworkTableEntry`` can still be used (for backwards compatibility), users are encouraged to migrate to use of type-specific Publisher/Subscriber/Entry classes as appropriate, or if necessary, ``GenericEntry``. It's important to note that unlike ``NetworkTableEntry``, these classes need to have appropriate lifetime management. Some functionality (e.g. persistent settings) has also moved to ``Topic`` properties.
```

Comment:
recommend adding links to the API docs

Reply:
Added in various other pages; added cross-links here to the new docs.